app/bundles/LeadBundle/Field/LeadFieldSaver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * @copyright   2018 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Field;
+
+use Mautic\LeadBundle\Entity\LeadField;
+use Mautic\LeadBundle\Entity\LeadFieldRepository;
+use Mautic\LeadBundle\Exception\NoListenerException;
+use Mautic\LeadBundle\Field\Dispatcher\FieldSaveDispatcher;
+
+class LeadFieldSaver
+{
+    /**
+     * @var LeadFieldRepository
+     */
+    private $leadFieldRepository;
+
+    /**
+     * @var FieldSaveDispatcher
+     */
+    private $fieldSaveDispatcher;
+
+    public function __construct(LeadFieldRepository $leadFieldRepository, FieldSaveDispatcher $fieldSaveDispatcher)
+    {
+        $this->leadFieldRepository = $leadFieldRepository;
+        $this->fieldSaveDispatcher = $fieldSaveDispatcher;
+    }
+
+    /**
+     * @param bool $isNew
+     */
+    public function saveLeadFieldEntity(LeadField $leadField, $isNew)
+    {
+        try {
+            $this->fieldSaveDispatcher->dispatchPreSaveEvent($leadField, $isNew);
+        } catch (NoListenerException $e) {
+        }
+
+        $this->leadFieldRepository->saveEntity($leadField);
+
+        try {
+            $this->fieldSaveDispatcher->dispatchPostSaveEvent($leadField, $isNew);
+        } catch (NoListenerException $e) {
+        }
+    }
+
+    public function saveLeadFieldEntityWithoutColumnCreated(LeadField $leadField)
```

Comment:
I didn't find any usage of this method.
I reported it in prev PR https://github.com/mautic/mautic/pull/6875#issuecomment-524740789
It was not working at that moment. Can somebody from Acquia check If It was push fully?

Reply:
It was. We use it internally but it's also used in https://github.com/mautic/mautic/pull/8828/files#diff-9f9e5870c9dceca25690ab685eecf62bc1ac712403fa87bd9c2791f1bda9b58dR58